core/storage/base_model/gae_models_test.py

```diff
@@ -357,6 +358,183 @@ def test_get_new_id_method_does_not_fail_with_bad_names(self):
         base_models.BaseModel.get_new_id({'a': 'b'})
 
 
+class TestBaseHumanMaintainedModel(base_models.BaseHumanMaintainedModel):
+    """Model that inherits the BaseHumanMaintainedModel for testing."""
+
+    pass
+
+
+class BaseHumanMaintainedModelTests(test_utils.GenericTestBase):
+    """Test the generic base human maintained model."""
+
+    MODEL_ID = 'model1'
+
+    def setUp(self):
+        super(BaseHumanMaintainedModelTests, self).setUp()
+        self.model_instance = TestBaseHumanMaintainedModel(id=self.MODEL_ID)
+        def mock_put(self):
+            self._last_updated_timestamp_is_fresh = True
+            self.last_updated_by_human = datetime.datetime.utcnow()
+
+            # These if conditions can be removed once the auto_now property
+            # is set True to these attributes.
+            if self.created_on is None:
+                self.created_on = datetime.datetime.utcnow()
+
+            if self.last_updated is None:
+                self.last_updated = datetime.datetime.utcnow()
+
+            super(base_models.BaseHumanMaintainedModel, self).put()
+
+        with self.swap(TestBaseHumanMaintainedModel, 'put', mock_put):
+            self.model_instance.put()
+
+    def test_put(self):
+        with self.assertRaisesRegexp(
+            NotImplementedError, 'Use put_for_human or put_for_bot instead'):
+            self.model_instance.put()
+
+    def test_put_for_human(self):
+        model_instance = TestBaseHumanMaintainedModel.get_by_id(self.MODEL_ID)
+        previous_last_updated_by_human = (
+            model_instance.last_updated_by_human)
+
+        model_instance.update_timestamps()
+        self.model_instance.put_for_human()
+        model_instance = TestBaseHumanMaintainedModel.get_by_id(self.MODEL_ID)
+
+        self.assertNotEqual(
+            previous_last_updated_by_human,
+            model_instance.last_updated_by_human)
```

Comment:
Could we do something like this?
```suggestion
        model_instance.update_timestamps()
        self.model_instance.put_for_human()
        new_model_instance = TestBaseHumanMaintainedModel.get_by_id(self.MODEL_ID)

        self.assertNotEqual(
            self.model_instance.last_updated_by_human,
            new_model_instance.last_updated_by_human)
```

Reply:
I have refactored it in a little bit different way. Done.